ksql-engine/src/main/java/io/confluent/ksql/codegen/SqlToJavaVisitor.java

```diff
@@ -574,7 +576,42 @@ private String visitBooleanComparisonExpression(final ComparisonExpression.Type
         final BetweenPredicate node,
         final Boolean unmangleNames
     ) {
-      throw new UnsupportedOperationException();
+      final Pair<String, Schema> value = process(node.getValue(), unmangleNames);
+      final Pair<String, Schema> min = process(node.getMin(), unmangleNames);
+      final Pair<String, Schema> max = process(node.getMax(), unmangleNames);
+
+      String expression = "(((Object) {value}) == null "
+          + "|| ((Object) {min}) == null "
+          + "|| ((Object) {max}) == null) "
+          + "? false "
+          + ": ";
+
+      switch (value.getRight().type()) {
+        case STRING:
+          expression += "({value}.compareTo({min}) >= 0 && {value}.compareTo({max}) <= 0)";
+          break;
+        case MAP:
+          throw new KsqlException("Cannot execute between with MAP values");
+        case ARRAY:
+          throw new KsqlException("Cannot execute between with ARRAY values");
+        case BOOLEAN:
+          throw new KsqlException("Cannot execute between with BOOLEAN values");
+        default:
+          expression += "{min} <= {value} && {value} <= {max}";
+          break;
```

Comment:
There is a BYTES type as well. is this the way to evaluate a BYTES column? 
Btw, wouldn't be better to use a default case to throw an exception in case a new type is added in the future? Like DATE, TIME, etc. That way we could throw the error as not supported instead of a bug because those types do not accept the <= evaluation.

Reply:
Actually I misunderstood what BYTES is - I'm not sure there's a good way to compare them by default (it falls more into the ARRAY category than scalar). Any suggestions? Also do we support it? I couldn't find any usage in the Visitor that alludes to BYTES.

Reply:
Actually, this test in `SchemaUtilTest` implies that we don't support it:
```
  @Test(expected = IllegalArgumentException.class)
  public void shouldThrowOnUnknownSchemaType() {
    SchemaUtil.getSchemaTypeAsSqlType(Schema.Type.BYTES);
  }
```

Reply:
ok, so we shouldn't use it in the between evaluator then :)

Reply:
adding bytes is tracked here: #1282